src/condor_schedd.V6/qmgmt.cpp

```diff
@@ -352,6 +352,46 @@ bool operator()(const prio_rec& a, const prio_rec& b) const
 }
 };
 
+// Return the same comparison as above, but just comparing the submitter
+// field.  This allows us to binary search the PrioRecArray by submitter
+// Note the comparison up to the submitter must match the above
+struct prio_rec_submitter_lb {
+bool operator()(const prio_rec& a, const std::string &user) const {
+	if (a.submitter.length() < user.length()) {
+		return false;
+	}
+
+	if (a.submitter.length() > user.length()) {
+		return true;
+	}
+
+	if (a.submitter < user) {
+		return false;
+	}
+
+	return false;
```

Comment:
This does not return the same as about when 'a.submitter > user'. The above function returns 'true', but this functions returns 'false'.

Reply:
Note that the order of user and a are reversed between the two functions. I believe the logic is correct. I'd recommend reversing the comparisons in prio_rec_submitter_ub, so that it more obviously matches prio_rec_submitter_lb and prio_compar.

Reply:
Ah. OK. Thank you for the clarification.

Reply:
Then why have the final comparison at all if the result will always be false after checking the length.